### PR TITLE
feat(combat): Lifeline implant — once-per-battle pre-hit threshold shield

### PR DESCRIPTION
**Pull request by TheSusort**
> Supersedes #158, which GitHub auto-closed when the H2/H3 base branch was deleted on merge. Rebased onto `main` (H1 #156 + H2/H3 #157 are now merged); content unchanged.

## Summary

Implements the **Lifeline** implant — the last remaining shield-grant source in sub-project H. When a *pure direct hit* would drop the carrier's HP below 30% of max HP, Lifeline grants a shield of `flatAmount(rarity) + 100% of the unit's attack` (capped at max HP) **before** the triggering hit's absorb step, so the rest of that hit drains shield→HP. The ship can still die if the hit exceeds the new shield + remaining HP. Once per battle.

Per-rarity flat amount: common 4000 / uncommon 6000 / rare 8000 / epic 10000 / legendary 12000.

### Mechanics (user-ratified)
- **Pre-hit / mid-hit timing** — shield goes up before the absorb, then the rest of the hit drains shield→HP (not a death-save; distinct from Cheat Death).
- **Downward crossing** — fires only when pre-hit HP ≥ 30% and the hit would bring it below 30%.
- **Direct damage only** — `byDirectDamage === true && bombPortion === 0`. DoTs (bypass shields) and bomb/detonation hits do **not** trigger it.
- **Once per battle** — combat-lifetime fired Set keyed `${victimId}:${abilityId}`.
- **Cap at max HP** — the total shield pool is capped at max HP.

### Implementation
- New `incoming-shield-grant` `AbilityConfig` type (+ minimal editor exhaustiveness stubs).
- Pure helper `src/utils/combat/thresholdShield.ts` (`thresholdShieldForHit`) — fully unit-tested incl. exact-boundary semantics.
- Engine intercept in `applyVictimDamage` (the single victim-side damage seam — covers both the aggregate/tank and positional paths), mirroring the D-PR3 `incoming-block` pattern; surfaces the granted delta to the H1 `perActorShield.granted` accumulator.
- `LIFELINE` registry entry + coverage tracker (3 spots + shape test).
- End-to-end integration tests through the real registry + `runCombat`.

### Architecture notes
- **Byte-identical** — no fixture equips Lifeline and no ship skill parses to `incoming-shield-grant`, so the intercept is inert for every existing actor; production goldens/snapshots are unchanged.
- Spec: `docs/superpowers/specs/2026-06-26-lifeline-shield-design.md`. Plan: `docs/superpowers/plans/2026-06-26-lifeline-shield.md`.
- **Strike correction:** the implant previously grouped with the shield-grant family is stat-only (flat attack) — no engine work; Lifeline is the only remaining shield-grant source.

> **Stacking:** based on `feat/combat-shield-system-h2-h3` (H2/H3 PR #157), which itself stacks on H1 PR #156. **Retarget this PR to `main` once the H stack (#156 → #157) merges.**

## Test Plan
- [x] Full suite green (3315 tests), goldens byte-identical (no `-u`)
- [x] `tsc --noEmit` clean, `npm run lint` clean (max-warnings 0)
- [x] `npm run audit:skills` → 141 ships, 0 findings (unchanged)
- [x] Helper unit tests incl. exact-boundary cases
- [x] Integration tests bite under mutation (intercept disabled → cases fail; restored → pass)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by TheSusort** [comment]
@coderabbitai review

File: src/utils/combat/engine.ts [hunk 1]
```diff
@@ -2773,6 +2781,46 @@ export function runCombat(input: CombatEngineInput): {
...
+                if (grant) {
+                    const newPool = Math.min(maxHp, shieldBefore + grant.grant);
+                    const granted = newPool - shieldBefore;
+                    victim.shieldPool = newPool;
+                    thresholdShieldFired.add(`${victim.id}:${grant.abilityId}`);
+                    // Surface the real pool growth on the H1 granted accumulator (StatCard).
+                    perActorShieldGranted.set(
+                        victim.id,
+                        (perActorShieldGranted.get(victim.id) ?? 0) + granted
+                    );
```
[the remark this answers is not in the document]
Comment (TheSusort) [reply src/utils/combat/engine.ts]: Fixed in 82c53269. `maxHp` here is `recipientMaxHp(victim.id)` (effective, can drop mid-combat via an HP-down debuff), so a pre-hit pool can legitimately exceed it. Now: `newPool = shieldBefore >= maxHp ? shieldBefore : min(maxHp, shieldBefore + grant)` and `granted = max(0, newPool - shieldBefore)` — only ever adds shield, never shrinks the pool. Byte-identical for all fixtures (none carry a pool above maxHp with a Lifeline ability).

File: src/constants/changelog.ts [hunk 2]
```diff
@@ -36,6 +36,7 @@ export const UNRELEASED_CHANGES: string[] = [
     'Combat simulator now models Block Buff — a unit affected by Block Buff can no longer gain new buffs while it lasts.',
     'Combat & DPS simulators now model two damage-over-time gear sets: Burner (applies Inferno for 2 turns when the ship attacks) and Decimation (+10% DoT damage per equipped set, up to +30%). Decimation now boosts your ship’s Inferno and Corrosion ticks in both the combat simulator and the DPS calculator.',
     'Combat simulator now models four shield sources: the Shield gear set grants the equipped ship a shield each turn (4% of its max HP), the Adaptive Plating implant grants the ship a shield from the damage it takes (once per round), the Abundant Renewal implant turns over-healing on an ally into a shield for that ally, and the Resonating Fury implant gives a chance to grant Crit Power Up to allies it shields.',
+    'Combat simulator now models the Lifeline implant: when a direct hit would drop a ship below 30% HP, it gains a shield (a flat amount plus 100% of its attack, capped at max HP) before the hit lands, so the rest of the hit drains shield then HP — once per battle.',
```
[the remark this answers is not in the document]
Comment (TheSusort) [reply src/constants/changelog.ts:39]: Fixed in 82c53269. Added '(capped at max HP)' to the common + uncommon Lifeline descriptions in implants.ts so all five rarities and the changelog describe the cap consistently (the engine caps every rarity).

File: src/components/skills/AbilityTypePicker.tsx [hunk 3]
```diff
@@ -26,6 +26,7 @@ const TYPE_LABELS: Record<AbilityType, string> = {
     control: 'Control',
     'incoming-reduction': 'Incoming Reduction',
     'incoming-block': 'Incoming Block',
+    'incoming-shield-grant': 'Incoming Shield Grant',
```
[the remark this answers is not in the document]
Comment (TheSusort) [reply src/components/skills/AbilityTypePicker.tsx:29]: Intentional — not a defect. `incoming-shield-grant` is an engine-internal ability type (a victim-side damage-path hook), like `incoming-block` and `incoming-reduction`. None of the three appear in `CATEGORIES`: they're in `TYPE_LABELS` for display only and are deliberately NOT user-pickable in the editor (the same convention the PR description calls out — 'editor stubs label-only, NOT user-pickable, like incoming-block'). Adding it to a CATEGORIES `types` array would expose a non-authorable type as a pickable button. Leaving as-is to match the established pattern.
